src/kOS/Screen/TermWindow.cs

```diff
@@ -625,21 +625,36 @@ public void ProcessOneInputChar(char ch, TelnetSingletonServer whichTelnet)
 
         void Type(char ch)
         {
-            if (shared != null && shared.Interpreter != null)
+            if (shared != null)
             {
-                shared.Interpreter.Type(ch);
-                if (IsOpen && keyClickEnabled)
-                    shared.SoundMaker.BeginSound("click");
+                if (shared.Interpreter != null && shared.Interpreter.isWaitingForCommand())
+                {
+                    shared.Interpreter.Type(ch);
+                    if (IsOpen && keyClickEnabled)
+                         shared.SoundMaker.BeginSound("click");
+                }
+                else 
+                {
+                    shared.Screen.CharInputQueue.Enqueue(ch);
+                }
             }
         }
 
         void SpecialKey(char key)
         {
```

Comment:
One consequence of putting the hook here is that the "cooking" of the input into more vanilla ascii has already happened in TermWindow.ProcessOneInputChar() at this point.  (Therefore people don't get UnicodeCommand.DELETELEFT.  Instead they get `0x0008` (control-H, the more vanilla ASCII way to say backspace).)  I'm not sure if this is good or bad.